sklearn/linear_model/ridge.py

```diff
@@ -1418,10 +1419,15 @@ def fit(self, X, y, sample_weight=None):
         -------
         self : object
         """
+
         X, y = check_X_y(X, y, ['csr', 'csc', 'coo'],
                          dtype=[np.float64],
                          multi_output=True, y_numeric=True)
 
+        # if classification, LabelBinarizer applied first in RidgeClassifierCV
+        # y (output of lb) would be -1 or 1 only
+        is_clf = ((y == -1) | (y == 1)).all()
```

Comment:
you can use `sklearn.utils.multiclass.type_of_target` instead and you will not need the comment.

Reply:
```suggestion
        is_clf = 'continuous' not in type_of_target(y)
```

Reply:
Thanks for the suggestion, this looks for more elegant ..

Reply:
@glemaitre 

unfortunately, it does not work for our case. When the values are discrete and having float values, then it returns multiclass.
```
type_of_target(array([1., 2., 3., ]))
'multiclass'
```

Reply:
But in a classification problem, you should not have floating values, isn't it?

Reply:
True. At the same time, regression problem might have only integers, which would be returned as `multiclass` by `type_of_target`. The above example could be valid for a regression problem as well know?

Reply:
regression problem will be represented as float because it should be a continuous variable.
This is enforced using `check_X_y` with `y_numeric=True`.

Reply:
let us consider the following dumb regression example:
```
>>> x, y = check_X_y([[1], [2], [3]], [1, 2, 3])
>>> type_of_target(y)
'multiclass'
```
hope you got the scenario now, which I am addressing. Thanks

Reply:
As there is no response from you, I am marking this as resolved. Please reopen it, if you find any conflict with it. Thanks

Reply:
couldn't this information be provided by a private function in the `_RidgeGCV` class that would be redefined in `RidgeClassifierCV`? this way it wouldn't need to be inferred from Y

Reply:
It doesn't seem to be a satisfactory solution. From y it's hard to tell the type of target. You can very well have a regression on `y=[1, 1, -1]`. 

Wouldn't just,
```py
is_clf = hasattr(self, "classes_")
```
work?

Reply:
My comment was about previous discussion, @jeromedockes 's suggestion could also work.

Reply:
> the error persists when `cv != None`

True. I had raised another [issue ](https://github.com/scikit-learn/scikit-learn/issues/14856) for this and am waiting for members to give me a thumbs up to start working on that.

Reply:
> couldn't this information be provided by a private function in the `_RidgeGCV` class that would be redefined in `RidgeClassifierCV`? this way it wouldn't need to be inferred from Y

Thanks for your inputs @jeromedockes, I think can we have it as one of the class params itself. Initially, I was with the impression that we are not supposed to change the class methods or params for this. I will make this change and commit it

Reply:
> a private function in the _RidgeGCV class that would be redefined in RidgeClassifierCV

RidgeClassifierCV does not inherit from _RidgeGCV.

`_RidgeGCV` is used in `_BaseRidgeCV` as a local estimator. 

I think we could simply pass a parameter `is_clf` to `_RidgeGCV` (@venkyyuvy it's OK to change the signature of classes that are private). 

Something like

```py
from ..base import is_classifier

## in _RidgeGCV.fit
estimator = _RidgeGCV(self.alphas,..., is_clf=is_classifier(self))

```

Reply:
> It doesn't seem to be a satisfactory solution. From y it's hard to tell the type of target. You can very well have a regression on `y=[1, 1, -1]`.
> 
> Wouldn't just,
> 
> ```python
> is_clf = hasattr(self, "classes_")
> ```
> 
> work?

Cann't agree more about your example. I was not able to come up with better solution before, without adding more methods or params.  
Sorry, I am not able to get your suggestion. Here, the `self` is `_ridgeGCV`, which is inherited my `LinearModel`. Can you please help me

Reply:
> I think we could simply pass a parameter is_clf to _RidgeGCV

+1 

> Sorry, I am not able to get your suggestion. Here, the self is _ridgeGCV, which is inherited my LinearModel. Can you please help me

Never mind, I looked at the class hierarchy a bit too quietly. Adding a init parameter is probably the way to go.

Reply:
Thanks. I will add a init param